tools/python/ptools/src/dftbplus_ptools/eigenvecout.py

```diff
@@ -0,0 +1,311 @@
+#------------------------------------------------------------------------------#
+#  DFTB+: general package for performing fast atomistic simulations            #
+#  Copyright (C) 2006 - 2022  DFTB+ developers group                           #
+#                                                                              #
+#  See the LICENSE file for terms of usage and distribution.                   #
+#------------------------------------------------------------------------------#
+
+"""
+Module for interaction with the eigenvec.out file.
+"""
+
+
+import re
+
+
+class Eigenvecout:
+    """class for reading eigenvectors"""
+
+    def __init__(self, filename="eigenvec.out"):
+        """Initialises the Eigenvecout class
+
+        Args:
+            filename (str): name of file
+        """
+        self._filename = filename
+
+
+    def read_eigenvec(self):
+        """function for reading eigenvector
+
+        Returns:
+            dictionary (dict/None): dictionary containing the eigenvector or
+                'None' if file not readable/found
+        """
+
+        try:
+            with open(self._filename, "r") as file:
+                lines = file.readlines()
+        except OSError:
+            return None
+
+        dictionary = {}
+        kpoint = None
+        eigennum = None
+        spin = None
+        atom = None
+        orbit = None
+
+        pattern1 = re.compile(r"""^K-point:\s+
+                              (?P<kpoint>[0-9]+)\s+
+                              Eigenvector:\s+
```

Comment:
```suggestion
                              Eigenvector:\s*
```
For a large number of eigenvectors (>999) this is a bug, since the current output of DFTB+ in eigenvec.out would look like this (note the missing white space between the colon and the index):
Eigenvector:1000